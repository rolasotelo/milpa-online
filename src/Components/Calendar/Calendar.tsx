import React from "react";
import { filter } from "underscore";
import useGameContext from "../../Hooks/useGameContext/useGameContext";
import MonthCard from "../MonthCard/MonthCard";

interface Props {}

const months = [
  "1 - Atlcahualo",
  "2 - Tlacaxipe-hualiztli",
  "3 - Tozoztontli",
  "4 - Huey tozoztli",
  "5 - Tóxcatl",
  "6 - Etzalcualiztli",
  "7 - Tecuilhuitontli",
  "8 - Huey tecuílhuitl",
  "9 - Tlaxochimaco-miccailhuitontli",
  "10 - Xocotlhuetzi-huey miccaílhuitl",
  "11 - Ochpaniztli",
  "12 - Teotleco",
  "13 - Tepeilhuitl",
  "14 - Quecholli",
  "15 - Panque-tzaliztli",
  "16 - Atemoztli",
];

const Calendar = (props: Props) => {
  const context = useGameContext();
  const currentCalendar = context.currentTurn
    ? filter(months, (month, index) => {
        return Math.abs(index - context.currentTurn!) <= 1;
      })
    : [];
  return (
    <div className="flex flex-wrap w-full h-20 bg-mexicanGreen-dark justify-between items-center px-1 rounded-xl mb-1 ">
      {currentCalendar.map((month: string, index) => {
        return (
          <MonthCard
            key={index}
            isCurrentMonth={index + 1 === context.currentTurn}
            text={month}
          />
        );
      })}
    </div>
  );
};

export default Calendar;
