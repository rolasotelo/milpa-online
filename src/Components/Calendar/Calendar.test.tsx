import React from "react";
import renderer from "react-test-renderer";

import Calendar from "./Calendar";

describe("<Calendar />", () => {
  const defaultProps = {};
  const wrapper = renderer.create(<Calendar {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
