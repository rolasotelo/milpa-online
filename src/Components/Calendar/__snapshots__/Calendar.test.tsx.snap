// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> render 1`] = `
<div
  className="flex w-full h-16 bg-mexicanGreen-dark"
>
  <div
    className="bg-mexicanGreen-light w-48 h-8"
  >
    ENERO
  </div>
  <div
    className="bg-mexicanBoneLight w-48 h-8"
  >
    FEBRERO
  </div>
  <div
    className="bg-mexicanBoneLight w-48 h-8"
  >
    MARZO
  </div>
  <div
    className="bg-mexicanBoneLight w-48 h-8"
  >
    ABRIL
  </div>
  <div
    className="bg-mexicanBoneLight w-48 h-8"
  >
    MAYO
  </div>
  <div
    className="bg-mexicanBoneLight w-48 h-8"
  >
    JUNIO
  </div>
  <div
    className="bg-mexicanBoneLight w-48 h-8"
  >
    JULIO
  </div>
  <div
    className="bg-mexicanBoneLight w-48 h-8"
  >
    AGOSTO
  </div>
  <div
    className="bg-mexicanBoneLight w-48 h-8"
  >
    SEPTIEMBRE
  </div>
  <div
    className="bg-mexicanBoneLight w-48 h-8"
  >
    OCTUBRE
  </div>
  <div
    className="bg-mexicanBoneLight w-48 h-8"
  >
    NOVIEMBRE
  </div>
  <div
    className="bg-mexicanBoneLight w-48 h-8"
  >
    DICIEMBRE
  </div>
</div>
`;
